src/ui/public/kuery/lib/functions/not.js

```diff
@@ -0,0 +1,38 @@
+import * as ast from '../ast';
+import { nodeTypes } from '../node_types';
+
+export function buildNodeParams(child, serializeStyle = 'operator') {
+  return {
+    arguments: [child],
+    serializeStyle
+  };
+}
+
+export function toElasticsearchQuery(node, indexPattern) {
+  let [ argument ] = node.arguments;
+  if (argument.type === 'literal') {
+    argument = nodeTypes.function.buildNode('is', '*', argument.value);
+  }
+
+  return {
+    bool: {
+      must_not: ast.toElasticsearchQuery(argument, indexPattern)
+    }
+  };
+}
+
+export function toKueryExpression(node) {
+  if (node.serializeStyle !== 'operator') {
+    throw new Error(`Cannot serialize "not" function as "${node.serializeStyle}"`);
+  }
+
+  const [ argument ] = node.arguments;
+  const queryString = ast.toKueryExpression(argument);
+
+  if (argument.function && (argument.function === 'and' || argument.function === 'or')) {
```

Comment:
I think you're right, this will need to become smarter eventually. I don't have the granularity required in `serializeStyle` for what you describe yet, and I'm not quite ready to dive into overhauling that. I think the best solution will become more clear to me over time, so I'd like to defer this until later.